src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +15,32 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        if (this.factory == null) {
+            throw new DataProcessingException("Factory not initialized");
+        }
+        Session session = this.factory.openSession();
+        if (session == null) {
+            throw new DataProcessingException("Session not initialized");
```

Comment:
Checking if 'session == null' is redundant since 'openSession()' should either return a valid session or throw an exception if it cannot do so.